lib/rubocop/formatter/formatter_set.rb

```diff
@@ -73,7 +76,7 @@ def add_formatter(formatter_type, output_path = nil)
           output = $stdout
         end
 
-        self << formatter_class.new(output)
+        self << formatter_class.new(output, @options)
```

Comment:
Adding a setter in the base class so you can call `self.last.options = @options` here would work, I believe. You would have to move any logic depending on options out of the formatter constructors, and put it somewhere else.